impl/src/main/java/org/ehcache/config/event/DefaultCacheEventListenerBuilder.java

```diff
@@ -0,0 +1,122 @@
+
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventFiring;
+import org.ehcache.event.EventOrdering;
+import org.ehcache.event.EventType;
+
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author rism
+ */
+public class DefaultCacheEventListenerBuilder {
+  private EventOrdering eventOrdering;
+  private EventFiring eventFiringMode;
+  private EnumSet<EventType> eventsToFireOn;
+  private Class<? extends CacheEventListener<?, ?>> listenerClass;
+
+  private DefaultCacheEventListenerBuilder() {
+  }
+
+  public static DefaultCacheEventListenerBuilder newEventListenerConfig(
+      Class<? extends CacheEventListener<?, ?>> listenerClass, EventType eventType, EventType... eventTypes){
+    DefaultCacheEventListenerBuilder listenerBuilder = new DefaultCacheEventListenerBuilder();
+    Set<org.ehcache.event.EventType> eventSetToFireOn = new HashSet<EventType>();
+    eventSetToFireOn.add(eventType);
+    for (EventType events : eventTypes) {
+      switch (events) {
+        case CREATED:
+          eventSetToFireOn.add(EventType.CREATED);
+          break;
+        case EVICTED:
+          eventSetToFireOn.add(EventType.EVICTED);
+          break;
+        case EXPIRED:
+          eventSetToFireOn.add(EventType.EXPIRED);
+          break;
+        case UPDATED:
+          eventSetToFireOn.add(EventType.UPDATED);
+          break;
+        case REMOVED:
+          eventSetToFireOn.add(EventType.REMOVED);
+          break;
+        default:
+          break;
+      }
+    }
+    listenerBuilder.eventsToFireOn = EnumSet.copyOf(eventSetToFireOn);
+    listenerBuilder.listenerClass = listenerClass;
+    return listenerBuilder;
+  }
+
+  public static DefaultCacheEventListenerBuilder newEventListenerConfig(
+      Class<? extends CacheEventListener<?, ?>> listenerClass,
+      Set<EventType> eventSetToFireOn) throws IllegalArgumentException {
+    if (eventSetToFireOn.isEmpty()) {
+      throw new IllegalArgumentException("EventType Set cannot be empty");
+    }
+    DefaultCacheEventListenerBuilder listenerBuilder = new DefaultCacheEventListenerBuilder();
+    listenerBuilder.eventsToFireOn = EnumSet.copyOf(eventSetToFireOn);
+    listenerBuilder.listenerClass = listenerClass;
+    return listenerBuilder;
+  }
+
+  public DefaultCacheEventListenerBuilder eventOrdering(EventOrdering eventOrdering){
+    this.eventOrdering = eventOrdering;
+    return this;
+  }
+
+  public DefaultCacheEventListenerBuilder ordered() {
+    this.eventOrdering = EventOrdering.ORDERED;
+    return this;
+  }
+
+  public DefaultCacheEventListenerBuilder unordered() {
+    this.eventOrdering = EventOrdering.UNORDERED;
+    return this;
+  }
+
+  public DefaultCacheEventListenerBuilder firingMode(EventFiring eventFiringMode){
+    this.eventFiringMode = eventFiringMode;
+    return this;
+  }
+
+  public DefaultCacheEventListenerBuilder synchronous(){
+    this.eventFiringMode = EventFiring.SYNCHRONOUS;
```

Comment:
Are we fine with ordered/unordered and synchrous/asynchronous naming or we want to go with firingMode and eventOrdering ?